### ISSUE TITLE
OMG compliance

### ISSUE THREAD

**Issue by PRADEEPKJ** [ADLINK-IST/opensplice#62]
Hi,

Does opensplice-dds comply with OMG specification Version 1.4? And RTPS specification 2.2?

Where can I find these information?

Thanks,
Pradeep

**Comment by hansvanthag** [comment]
yes it does, see http://www.prismtech.com/vortex/vortex-opensplice

**Comment by WOLFY888** [comment]
> yes it does, see http://www.prismtech.com/vortex/vortex-opensplice

While working with version 6.9, I saw an implementation mismatch with the OMG standard. According to the OMG version 1.4 a Topic and a DataWriter must have a durability_service QosPolicy(page 93), but in the implementation documentation (OpenSplice_refman_CPP.pdf, page 42) this QosPolicy is available only for the Topic, the DataWriter does not have such a policy. In this situation I am mistaken or such need is really dictated by something?

**Comment by kortstee** [comment]
@WOLFY888 The DurabilityServiceQosPolicy is meant to drive the durability behaviour for a specific Topic and that makes sense in a data-centric world like DDS where there's basically one global data space. Therefore it does not make sense to set it per DataWriter as well. How would the DDS implementation have to deal with different DurabilityQosPolicy settings for a given Topic, right? The availability of this policy on the DataWriter is therefore deemed an omission in the DDS standard. There is already two open issues in the OMG DDS Revision Task Force; https://issues.omg.org/issues/lists/dds-rtf5#issue-13958 and https://issues.omg.org/issues/lists/dds-rtf5#issue-14092 that propose to remove that policy from the DataWriter. As it would not be possible to implement unambiguous semantics with this policy on DataWriter-level, OpenSplice does not provide them there. Makes sense?
